### PR TITLE
Simplify CreatePackageReport and suppressed plotting

### PR DESCRIPTION
**Pull request by jameslamb**
This PR clarifies the way package reports are created. Changes:

1) makes it clear that `RenderPackageReport` is internal by removing roxygen docs and adding `.`
2) suppresses plotting in the calling session, making it clear that plots only get generated in the report. This will avoid a few annoyances on the user side, like things breaking if you run inside RStudio and have your plotting window minimized
3) removes logger code in `RenderPackageReport` and creates reusable `silence_logger` and `unsilence_logger` functions
4) simplifies the way we fit reporters to a packageName / packagePath

### CODE REVIEW

**Comment by jameslamb** [comment]
@bburns632 I'll rebase and update this

File: R/CreatePackageReport.R [hunk 1]
```diff
@@ -33,59 +35,54 @@ CreatePackageReport <- function(packageName
         log_fatal(msg)
     }
     
-    log_info(paste0("Creating package report for package "
-                   , packageName
-                   , " with reporters:\n\n"
-                   , paste(unlist(lapply(packageReporters, function(x) class(x)[1]))
-                           , collapse = "\n")))
+    # Fit reporters with package name and (maybe) package path
+    reporters <- lapply(
```
Comment (bburns632) [inline R/CreatePackageReport.R]: smooth.

File: R/CreatePackageReport.R [hunk 2]
```diff
@@ -33,59 +35,54 @@ CreatePackageReport <- function(packageName
...
-        reporter$set_package(packageName, packagePath)
-        
-        reporter$calculate_all_metrics()
-        # TODO: replace plot_network() with render_report() which is then rendered into a parent report.
-        reporter$plot_network()
-        log_info(paste("Done Package Reporter",class(reporter)[1]))
-    }
-    
-    RenderPackageReport(
+    .RenderPackageReport(
```
Comment (bburns632) [inline R/CreatePackageReport.R]: Good Call.  We also need to discuss what functions are displayed in pkgdown website.

**Comment by jameslamb** [comment]
rebased 👍

**Comment by jameslamb** [comment]
@bburns632 idk why this is breaking on Travis. It just has the dread "could not install package" error. Will fix it right now.

Do you approve of the changes? Or were you saying we need to discuss pkgdown stuff before this can get merged

**Comment by jameslamb** [comment]
Fixed! I was importing `futile.logger::threshold` (which doesn't exist) not `futile.logger::flog.threshold`

**Comment by jameslamb** [comment]
@bburns632 I am going to merge this. It's building on travis and the changes are noncontroversial.

**Pull request merged**